src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesService.java

```diff
@@ -119,7 +120,7 @@ private ResponseEntity<Object> createOverseasEntity(Transaction transaction,
 
         // Update company name set on the transaction and add a link to our newly created Overseas Entity
         // submission (aka resource) to the transaction (and potentially also a link for the 'resume' journey)
-        updateTransactionWithLinksAndCompanyName(transaction, overseasEntitySubmissionDto.getEntityName(), submissionId,
+        updateTransactionWithLinksAndCompanyName(transaction, overseasEntitySubmissionDto.getEntityName().getName(), submissionId,
```

Comment:
this could throw a null pointer if the submission dto doesn't have EntityName block (it is optional in our partial validation).
I'd check the name before calling updateTransactionWithLinksAndCompanyName
eg
var entityName = null;
if (Objects.nonNull(overseasEntitySubmissionDto.getEntityName())) {
  entityName = overseasEntitySubmissionDto.getEntityName().getName();
}
updateTransactionWithLinksAndCompanyName(transaction, entityName, submissionId,...

Reply:
actually, my suggestion my also have an edge case.
Instead of my previous suggestion, I'd change the updateTransactionWithLinksAndCompanyName method to take the EntityNameDto as a param instead of the companyNameString
Then, in updateTransactionWithLinksAndCompanyName

change
        transaction.setCompanyName(companyName);

to be 
if (Objects.nonNull(entityNameDto)) {
          transaction.setCompanyName(entityNameDto.getName());
}


this way we will only set the name if it is present.
my previous suggestion would set to null if that block was missing, but the name could have been set by a previous call to PUT, and if a second call to PUT is made and entityNAme block is missing, that is not necessarily an instruction to set the name to null.

I'll include @mwestacott , might need to discuss this

Reply:
Updated to set to null first and then update the value if we have the block.